### PR TITLE
Markdown representations of Jupyter notebooks

### PR DESCRIPTION
**Pull request by jmarshrossney**
This PR proposes to keep human-readable Markdown representations of Jupyter notebooks under version control, instead of the notebooks themselves. This makes it easier to track changes and reduces the chance of accidentally committing a bunch of generated outputs, which can increase the size of the repository for not necessarily good reasons.

The mechanism through which `.ipynb` and `.md` versions of the notebooks are synchronised is a package called [Jupytext](https://jupytext.readthedocs.io/en/latest/).

There are various fancy things one can do, including a pre-commit hook to sync notebooks, but perhaps the nice thing about this is the amount of extra effort required to implement this is tiny. You just need to remember to run `jupytext --sync notebooks/*` before committing - that's it!

This is not to suggest that we should never commit full notebooks with computed outputs, e.g. for showcasing results down the line.

### CODE REVIEW

**Comment by jmarshrossney** [comment]
Note - may need rebasing if #13 doesn't get merged.

**Comment by metazool** [comment]
Opinions welcomed on this from the rest of @NERC-CEH/eds-rse - I like the idea of having meaningful diffs, as notebook-based research code can present an awkward barrier to collaboration, but slightly reserved about adding extra workflow steps

**Comment by metazool** [comment]
https://github.com/maid/maid/issues/324 i think here's the bug in the coverage action from forks with a link to discussion about mitigating it, but my suggested resolution is bypassing this in favour of making branches in this repo directly

File: README.md [hunk 1]
```diff
@@ -42,6 +41,20 @@ Get started by cloning this repository and running
...
+### Running Jupyter notebooks
+
+The `notebooks/` directory contains Markdown (`.md`) representations of the notebooks.
+To create Jupyter notebooks (`.ipynb`), run the following command with the conda environment activated:
+
+```sh
+jupytext --sync notebooks/*
+```
+
+If you modify the contents of a notebook, run the command after closing the notebook to re-sync the `.ipynb` and `.md` representations before committing.
```
Comment (metazool) [inline README.md:53]: I brought up this use of `jupytext` in the daily catchup @jmarshrossney and no opinions were forthcoming, so I'm going to call this a worthwhile contribution and try adopting it as a routine

**Review by metazool — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jmarshrossney) [reply README.md:53]: Thanks Jo - I appreciate it's a slightly annoying additional step. One nice thing is that it's completely trivial to change our minds and go back to tracking the `.ipynb` files down the line!

**Comment by jmarshrossney** [comment]
> [maid/maid#324](https://github.com/maid/maid/issues/324) i think here's the bug in the coverage action from forks with a link to discussion about mitigating it, but my suggested resolution is bypassing this in favour of making branches in this repo directly

Interesting, and noted.

**Pull request merged**